### ISSUE TITLE
Different output when running credo with the `--read-from-stdin` option

### ISSUE THREAD

**Issue by GiovanniHessel94** [rrrene/credo#1082]
Hey I'm having an issue and I couldn't find anything related to it. Sorry if it has already being addressed somewhere else.

### Environment

* Credo version (`mix credo -v`):

`1.7.1-ref.main.eb7b5a4800+uncommittedchanges`

* Erlang/Elixir version (`elixir -v`):

```
Erlang/OTP 24 [erts-12.3.2] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit]

Elixir 1.14.5 (compiled with Erlang/OTP 24)
```

* Operating system:

```
Versão do Windows: 10.0.22621.2428
Versão do WSL: 1.2.5.0
```

### Expected outcome

I expected to have the same output when running the commands `mix credo filename.ex --strict` and `mix credo --read-from-stdin < filename.ex --strict` in the same file.

### Actual outcome

It seems that with the `--read-from-stdin` option it detects the `readability:Credo.Check.Readability.ModuleDoc` while without the option it doesn't.

![image](https://github.com/rrrene/credo/assets/26199302/d5a50dfd-4e88-4292-8075-144379464341)

**Comment by rrrene** [comment]
Thx for reporting this :+1:

Unfortunately, I can not reproduce this on my machine. One thing I noticed is that the `--strict` is misplaced:

```bash
mix credo --read-from-stdin < filename.ex --strict
```

should probably be

```bash
mix credo --strict --read-from-stdin < filename.ex
```

Could you check that?

**Comment by GiovanniHessel94** [comment]
@rrrene sorry for the delay, tried that and the output is the same (as It should probably be).

I think "the problem" is in [this](https://github.com/rrrene/credo/blob/d1d76ea4c1b1bfed0f3ef9835cde070867e9328e/lib/credo/check/readability/module_doc.ex#L52) line, it only happens in the `test` files, as per the example, and those are `.exs` so it returns an empty list when running the `mix credo` command, now when we use the `--read-from-stdin` the `filename` comes as `stdin` so it applies the rules.

Not sure if something should be done about it, but that seems to be the reason of the different output using this option, and as I'm using the vsCode extension and it uses the  `--read-from-stdin` option, this warning is always present in the `test` files.

**Comment by rrrene** [comment]
@GiovanniHessel94 ah, you are right. :+1:

[The filename can be passed](https://hexdocs.pm/credo/cli_switches.html#parsing-source-from-stdin) and it seems that the integration you are using is not making use of that feature.

**Comment by GiovanniHessel94** [comment]
I'll open an issue in their repo, so I think we can close this one. Thanks!
